third_party/move/move-compiler-v2/tests/checking/specs/expressions_err.exp

```diff
@@ -12,37 +12,37 @@ error: no function named `not_declared` found
 15 │       not_declared() // Undeclared function.
    │       ^^^^^^^^^^^^^^
 
-error: expected `num` but found `bool`
+error: cannot return `bool` from a function with result type `num`
    ┌─ tests/checking/specs/expressions_err.move:19:7
    │
 19 │       false // Wrong result type.
    │       ^^^^^
 
-error: invalid call of `>`: expected `num` but found `vector<num>` for argument 1
+error: cannot use `vector<num>` with an operator which expects a value of type `num`
    ┌─ tests/checking/specs/expressions_err.move:23:7
    │
 23 │       x > y // No matching function.
    │       ^
 
-error: expected `(num, bool)` but found `bool`
+error: cannot return `bool` from a function with result type `(num, bool)`
    ┌─ tests/checking/specs/expressions_err.move:27:7
    │
 27 │       false // Wrong result type tuple.
    │       ^^^^^
 
-error: invalid call of `M::wrongly_typed_callee`: expected `bool` but found `u256` for argument 2
+error: cannot use `u256` with an operator which expects a value of type `bool`
    ┌─ tests/checking/specs/expressions_err.move:32:31
    │
 32 │       wrongly_typed_callee(1, 1) // Wrongly typed function application
    │                               ^
 
-error: invalid call of `M::wrongly_typed_fun_arg_callee`: expected `num` but found `bool` for argument 1
+error: cannot use `|num|bool` with an operator which expects a value of type `|num|num`
```

Comment:
Nitpick: This error could use the word "type" somewhere before `|num|bool` to make it very clear that it's not an expression.  Perhaps the format pattern could be
```
"value of type `{}` provided for an argument of type `{}`"
```
which would avoid the call/function/operator distinction.

Reply:
As I said, pleass go and change all those messages async. I assume you have seen the one central file which defines them now (`ty.rs`).